src/Microsoft.AspNet.Mvc.Razor/DependencyInjection/MvcRazorMvcBuilderExtensions.cs

```diff
@@ -53,13 +54,17 @@ public static class MvcRazorMvcBuilderExtensions
 
             return builder;
         }
-        
+
         public static IMvcBuilder AddPrecompiledRazorViews(
             [NotNull] this IMvcBuilder builder,
             [NotNull] params Assembly[] assemblies)
         {
-            var razorFileInfos = RazorFileInfoCollections.GetFileInfoCollections(assemblies);
-            builder.Services.TryAddEnumerable(ServiceDescriptor.Instance(razorFileInfos));
+            builder.Services.Replace(
+                ServiceDescriptor.Singleton<ICompilerCacheProvider>(serviceProvider =>
+                    ActivatorUtilities.CreateInstance<PrecompiledViewsCompilerCacheProvider>(
+                        serviceProvider,
+                        assemblies.AsEnumerable())));
```

Comment:
`ActivatorUtilities` doesn't like it when we expect `IEnumerable<>` but pass in `Assembly[]`.